tutorials/auto-data-tokenize/index.md

```diff
@@ -0,0 +1,446 @@
+   The `keySchema` shows a mapping between the leaf value's key's to a schema key to demonstrate that leaf-nodes of same type share the same key-schema, for example: `$.contacts[0].contact.number` is logically same as `$.contacts[1].contact.number` as both of them have the same key-schema `$.contacts.contact.number`.
+   
+   ```json
+   {
+      "values": {
+       "$.name": "Jane Doe",
+       "$.contacts[0].contact.type": "WORK",
+       "$.contacts[0].contact.number": 2124567890,
+       "$.contacts[1].contact.type": "WORK",
+       "$.contacts[1].contact.number": 5304321234   
+      },
+   
+      "keySchema": {
+       "$.name": "$.name",
+       "$.contacts[0].contact.type": "$.contacts.contact.type",
+       "$.contacts[0].contact.number": "$.contacts.contact.number",
+       "$.contacts[1].contact.type": "$.contacts.contact.type",
+       "$.contacts[1].contact.number": "$.contacts.contact.number"   
+      }
+   }
+   ```
+
+## Prerequisites
+
+This tutorial assumes some familiarity with shell scripts and basic knowledge of Google Cloud.
+
+## Objectives
+
+1. Understand record sampling and identify sensitive columns using DLP.
+1. Use of symmetric encryption to tokenize data using KMS wrapped data-encryption key.
+
+## Costs
+
+This tutorial uses billable components of Google Cloud, including the following:
+
+* [Dataflow](https://cloud.google.com/dataflow/pricing)
+* [Cloud Storage](https://cloud.google.com/storage/pricing)
+* [Cloud Data Loss Prevention](https://cloud.google.com/dlp/pricing)
+* [Cloud KMS](https://cloud.google.com/kms/pricing)
+
+Use the [pricing calculator](https://cloud.google.com/products/calculator) to generate a cost estimate based on your
+projected usage.
+
+## Before you begin
+
+For this tutorial, you need a Google Cloud [project](https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy#projects). To make
+cleanup easiest at the end of the tutorial, we recommend that you create a new project for this tutorial.
+
+1. [Create a Google Cloud project](https://console.cloud.google.com/projectselector2/home/dashboard).
+1. Make sure that [billing is enabled](https://support.google.com/cloud/answer/6293499#enable-billing) for your Google
+   Cloud project.
+1. [Open Cloud Shell](https://console.cloud.google.com/?cloudshell=true).
+
+   At the bottom of the Cloud Console, a [Cloud Shell](https://cloud.google.com/shell/docs/features) session opens and
+   displays a command-line prompt. Cloud Shell is a shell environment with the Cloud SDK already installed, including
+   the [gcloud](https://cloud.google.com/sdk/gcloud/) command-line tool, and with values already set for your current
+   project. It can take a few seconds for the session to initialize.
+
+1. Enable APIs for Data Catalog, BigQuery, Pub/Sub, Dataflow, and Cloud Storage services with the following command:
```

Comment:
should we match the product names with the command line below? seems a few products (e.g., Cloud DLP and Cloud KMS) are missing in this sentence

Reply:
ACK. Done